### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by P1200**
Basic functionality od registration and login

### CODE REVIEW

File: src/main/java/com/candyshop/islodycze/registration/CustomUserDetails.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Collection;
+
+import com.candyshop.islodycze.model.User;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+public class CustomUserDetails implements UserDetails {
+
+    private User user;
+
```
Comment (Amaimon208) [inline src/main/java/com/candyshop/islodycze/registration/CustomUserDetails.java:13]: Do we need all functions like isCredentialsNonExpired?

**Review by Amaimon208 — APPROVED** [review]

**Review by emularczyk — APPROVED** [review]

File: src/main/java/com/candyshop/islodycze/registration/UserRepository.java [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 import com.candyshop.islodycze.model.User;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface UserRepository extends JpaRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.email = ?1")
+    User findByEmail(String email);
```
Comment (Amaimon208) [inline src/main/java/com/candyshop/islodycze/registration/UserRepository.java:8]: Did you need to use Query? Wouldn't custom data JPA function like:
  List<User> findByEmail(String email);
 be enought?

File: src/main/java/com/candyshop/islodycze/registration/CustomUserDetails.java (hunk 1 above)
Comment (P1200) [reply src/main/java/com/candyshop/islodycze/registration/CustomUserDetails.java:13]: Implementation of these methods is forced by interface, they have to stay

File: src/main/java/com/candyshop/islodycze/registration/UserRepository.java (hunk 2 above)
Comment (P1200) [reply src/main/java/com/candyshop/islodycze/registration/UserRepository.java:8]: Right, i removed it

**Pull request merged**